c++/include/orc/Writer.hh

```diff
@@ -0,0 +1,294 @@
+     * @return if not set, return default value which is speed.
+     */
+    CompressionStrategy getCompressionStrategy() const;
+
+    /**
+     * Set the writer version.
+     */
+    WriterOptions& setWriterVersion(WriterVersion version);
+
+    /**
+     * Get the strip size.
+     * @return if not set, return default value which is WriterVersion_ORIGINAL
+     */
+    WriterVersion getWriterVersion() const;
+
+    /**
+     * Set the padding tolerance.
+     */
+    WriterOptions& setPaddingTolerance(double tolerance);
+
+    /**
+     * Get the padding tolerance.
+     * @return if not set, return default value which is zero.
+     */
+    double getPaddingTolerance() const;
+
+    /**
+     * Set the memory pool.
+     */
+    WriterOptions& setMemoryPool(MemoryPool * memoryPool);
+
+    /**
+     * Get the strip size.
+     * @return if not set, return default memory pool.
+     */
+    MemoryPool * getMemoryPool() const;
+
+    /**
+     * Set the error stream.
+     */
+    WriterOptions& setErrorStream(std::ostream& errStream);
+
+    /**
+     * Get the error stream.
+     * @return if not set, return std::err.
+     */
+    std::ostream * getErrorStream() const;
+
+    /**
+     * Set whether or not to write statistics (file statistics,
+     * stripe statistics, etc.)
+     */
+    WriterOptions& setEnableStats(bool enable);
+
+    /**
+     * Get whether or not to write statistics (file statistics,
+     * stripe statistics, etc.)
+     * @return if not set, the default is true
+     */
+     bool getEnableStats() const;
+
+    /**
+     * Set whether or not to update string min/max stats
+     */
+    WriterOptions& setEnableStrStatsCmp(bool enable);
+
+    /**
+     * Get whether or not to update string min/max stats
+     * @return if not set, the default is false
+     */
+    bool getEnableStrStatsCmp() const;
+
+    /**
+     * Set whether or not to write row group index
+     */
+    WriterOptions& setEnableIndex(bool enable);
+
+    /**
+     * Get whether or not to write row group index
+     * @return if not set, the default is false
+     */
+    bool getEnableIndex() const;
+
+    /**
+     * Get writer timezone
+     * @return writer timezone
+     */
+    const Timezone* getTimezone() const;
+
+    /**
+     * Set writer timezone
+     * @param zone writer timezone name
+     */
+    WriterOptions& setTimezone(const std::string& zone);
+  };
+
+  class Writer {
+  public:
+    virtual ~Writer();
+
+    /**
+     * Create a row batch for writing the columns into this file.
+     * @param size the number of rows to read
+     * @return a new ColumnVectorBatch to write into
+     */
+    virtual ORC_UNIQUE_PTR<ColumnVectorBatch> createRowBatch(uint64_t size
+                                                             ) const = 0;
+
+    /**
+     * Add a row batch into current writer.
+     * @param rowsToAdd the row batch data to write.
+     */
+    virtual void add(ColumnVectorBatch& rowsToAdd) = 0;
+
```

Comment:
Will be helpful to have `virtual bool canAdd(ColumnVectorBatch& rowsToAdd) = 0;` return true if there is enough buffer available to add the ColumnVectorBatch, false otherwise.